src/libcollections/vec.rs

```diff
@@ -699,16 +699,7 @@ impl<T> Vec<T> {
     #[inline]
     #[stable(feature = "rust1", since = "1.0.0")]
     pub fn push(&mut self, value: T) {
-        // This will panic or abort if we would allocate > isize::MAX bytes
-        // or if the length increment would overflow for zero-sized types.
-        if self.len == self.buf.cap() {
-            self.buf.double();
-        }
-        unsafe {
-            let end = self.as_mut_ptr().offset(self.len as isize);
-            ptr::write(end, value);
-            self.len += 1;
-        }
+        self <- value;
```

Comment:
I had some benchmarks implemented in https://github.com/nagisa/rust/commit/7b1b36da999139bbd2cff4b957f2042239ff6247 but they’re not enough at all to properly test all the cases (e.g. large-sized `T`).